src/client.rs

```diff
@@ -991,6 +1061,86 @@ pub struct ClientConfig {
     /// A list of extra root certificate to trust. This can be used to connect
     /// to servers using self-signed certificates
     pub extra_root_certificates: Vec<Certificate>,
+
+    /// A function that defines the client's behaviour if an Image Index Manifest
+    /// (i.e Manifest List) is encountered when pulling an image.
+    /// Defaults to [current_platform_resolver](self::current_platform_resolver),
+    /// which attempts to choose an image matching the running OS and Arch.
+    ///
+    /// If set to None, an error is raised if an Image Index manifest is received
+    /// during an image pull.
+    pub platform_resolver: Option<Box<PlatformResolverFn>>,
+}
+
+impl Default for ClientConfig {
+    fn default() -> Self {
+        Self {
+            protocol: ClientProtocol::default(),
+            accept_invalid_hostnames: false,
+            accept_invalid_certificates: false,
+            extra_root_certificates: Vec::new(),
+            platform_resolver: Some(Box::new(current_platform_resolver)),
+        }
+    }
+}
+
+type PlatformResolverFn = dyn Fn(&[ImageIndexEntry]) -> Option<String>;
```

Comment:
Instead of this all being in the client package, perhaps this would be useful in a separate module. Doesn't have to be done in this PR, but just thought this would be useful outside of the context of the OCI client.